vochain/transaction.go

```diff
@@ -563,106 +560,165 @@ func (app *BaseApplication) VoteEnvelopeCheck(ve *models.VoteEnvelope, txBytes,
 }
 
 // AdminTxCheck is an abstraction of ABCI checkTx for an admin transaction
-func AdminTxCheck(vtx *models.Tx, txBytes, signature []byte, state *State) error {
+func AdminTxCheck(vtx *models.Tx, txBytes, signature []byte, state *State) (common.Address, error) {
 	if vtx == nil {
-		return ErrNilTx
+		return common.Address{}, ErrNilTx
 	}
 	tx := vtx.GetAdmin()
 	// check signature available
 	if signature == nil || tx == nil || txBytes == nil {
-		return fmt.Errorf("missing signature and/or admin transaction")
+		return common.Address{}, fmt.Errorf("missing signature and/or admin transaction")
 	}
-	// get oracles
-	oracles, err := state.Oracles(false)
-	if err != nil || len(oracles) == 0 {
-		return fmt.Errorf("cannot check authorization against a nil or empty oracle list")
+
+	pubKey, err := ethereum.PubKeyFromSignature(txBytes, signature)
+	if err != nil {
+		return common.Address{}, fmt.Errorf("cannot extract public key from signature: %w", err)
+	}
+	addr, err := ethereum.AddrFromPublicKey(pubKey)
+	if err != nil {
+		return common.Address{}, fmt.Errorf("cannot extract address from public key: %w", err)
 	}
+	log.Debugf("checking admin signed tx %+v by addr %x", tx, addr)
 
-	if authorized, addr, err := verifySignatureAgainstOracles(
-		oracles, txBytes, signature); err != nil {
-		return err
-	} else if !authorized {
-		return fmt.Errorf("unauthorized to perform an adminTx, address: %s", addr.Hex())
+	// only treasurer can do admin txs
+	// oracles (keykeepers) can still add and reveal process keys
+	if tx.Txtype != models.TxType_ADD_PROCESS_KEYS && tx.Txtype != models.TxType_REVEAL_PROCESS_KEYS {
+		// get treasurer
+		treasurer, err := state.Treasurer(false)
```

Comment:
Better this code in a function to check if the address is treasurer, it might be reused